egs/librispeech/asr/simple_v1/mmi_bigram_embeddings_train.py

```diff
@@ -132,7 +132,9 @@ def get_objf(batch: Dict,
     dense_fsa_vec = k2.DenseFsaVec(nnet_output, supervision_segments)
     assert nnet_output.device == device
 
+    logging.info('the following crashes in the second batch (start)')
     num_lats = k2.intersect_dense(num_graph, dense_fsa_vec, 10.0)
```

Comment:
The training crashes at the second batch in this line.
<img width="1357" alt="Screen Shot 2021-03-05 at 12 39 24 PM" src="https://user-images.githubusercontent.com/5284924/110068120-fbab7a00-7daf-11eb-9a24-6a3839aadcad.png">


It is strange that the first batch works fine. All the intermediate FSAs from the previous batch should be destroyed
and every batch starts with num_graph and den_graph recreated. I don't understand why it crashes at the
second batch.

If I comment out 
```python
tot_score = tot_score + second_pass_tot_score
```

that is, discarding the computation result of the second pass, the training continues just like without the second pass.

Reply:
Possibly the model somehow got NaN's in it?  You could check the model output for NaN's by computing its sum.

Reply:
thanks, looking into it.

Reply:
I find the reason. `pathphone_idx`'s for the arcs that entering the final state have no corresponding `nnet_output`
so their embeddings contain `[0, -inf, -inf, ...., -inf]`. 

How can we handle these `pathphone_idx`'s?  What embeddings should be assigned to them? @danpovey

Reply:
Ah.  I suggest to use zeros.